src/cli-validator/runValidator.js

```diff
@@ -288,6 +288,33 @@ const processInput = async function(program) {
       results.hint = false;
     }
 
+    // fail on errors or if number of warnings exceeds warnings limit
+    if (results.error) {
+      exitCode = 1;
+    } else {
+      // Calculate number of warnings and set exit code to 1 if warning limit exceeded
+      let numWarnings = 0;
+      for (const key of Object.keys(results.warnings)) {
+        numWarnings += results.warnings[key].length;
+      }
+      if (numWarnings > limitsObject.warnings) {
+        exitCode = 1;
+        // add the exceeded warnings limit as an error
+        if (!results.errors) {
+          results.errors = {};
+        }
+        results.errors['warnings-limit'] = [
+          {
+            path: [],
+            message: `Number of warnings (${numWarnings}) exceeds warnings limit (${
+              limitsObject.warnings
+            }).`,
+            rule: 'warnings_limit'
```

Comment:
I may be overthinking this but I wonder if this isn't the most clear. We don't actually have a rule anywhere called "warnings_limit" so I don't want to mislead users into thinking that they should be setting that property somewhere. I suppose that if they see this message, they already know how to configure it, and I don't have a better option. Perhaps this error doesn't need a "rule" though.

Reply:
I think you're right. This would be better suited as "builtin"